src/halmos/build.py

```diff
@@ -0,0 +1,174 @@
+import json
```

Comment:
no changes here, just moving build-related functionality out of `__main__.py` and into its own module